### PR TITLE
Add support for compiles on HPE NonStop

### PR DESCRIPTION
**Pull request by rsbeckerca**
Fixes: #229

### CODE REVIEW

**Comment by rsbeckerca** [comment]
I still need help convincing `configure` to build a DLL. The arguments are not the same as for gcc and configure gets confused about it. `-Wshared` is the trigger to build `.so` files.

**Comment by eli-schwartz** [comment]
Looking at the relevant libtool logic it appears that it matches `irix5* | irix6* | nonstopux*` and defines `$archive_cmds` to use `-shared`. Rough summary:

```
      if test yes = "$GCC"; then
        archive_cmds='$CC -shared $pic_flag $libobjs $deplibs $compiler_flags $wl-soname $wl$soname `test -n "$verstring" && func_echo_all "$wl-set_version $wl$verstring"` $wl-update_registry $wl$output_objdir/so_locations -o $lib'
      else
        archive_cmds='$CC -shared $libobjs $deplibs $compiler_flags -soname $soname `test -n "$verstring" && func_echo_all "-set_version $verstring"` -update_registry $output_objdir/so_locations -o $lib'
```

**Comment by rsbeckerca** [comment]
> Looking at the relevant libtool logic it appears that it matches `irix5* | irix6* | nonstopux*` and defines `$archive_cmds` to use `-shared`. Rough summary:
> 
> ```
>       if test yes = "$GCC"; then
>         archive_cmds='$CC -shared $pic_flag $libobjs $deplibs $compiler_flags $wl-soname $wl$soname `test -n "$verstring" && func_echo_all "$wl-set_version $wl$verstring"` $wl-update_registry $wl$output_objdir/so_locations -o $lib'
>       else
>         archive_cmds='$CC -shared $libobjs $deplibs $compiler_flags -soname $soname `test -n "$verstring" && func_echo_all "-set_version $verstring"` -update_registry $output_objdir/so_locations -o $lib'
> ```

`nonstopux*` is not the same. It is, in fact, deprecated - old product naming from when Compaq owned Tandem and made everything "NonStop". This is a separate set of platforms - `nsx-tandem-nsk` and `nse-tandem-nsk` are currently value entries from `config.guess`. `-Wshared` is what is needed.

**Comment by eli-schwartz** [comment]
:( 

Those host identifiers aren't shown *anywhere* in libtool.m4 (which is used to generate the libtool script by configure) that I can tell.

My conclusion is that while config.guess knows the platform exists, libtool in particular doesn't actually do anything whatsoever with that information and does not know how to build libraries for nonstop. (This is ironic since the one selling point of libtool is supposed to be that it works on any platform ever.)

**Comment by rsbeckerca** [comment]
Explains why libtool is a noop on the platform

**Comment by rsbeckerca** [comment]
This PR also includes changes to some tests to honor `HAVE_ALLOCA_H`. The changes build and test correctly on the NonStop (TANDEM) x86 platform.

File: tests/test-is-cookie-domain-acceptable.c [hunk 1]
```diff
@@ -131,7 +131,11 @@ int main(int argc, const char * const *argv)
 
 		if (valgrind && *valgrind) {
 			size_t cmdsize = strlen(valgrind) + strlen(argv[0]) + 32;
+#ifndef HAVE_ALLOCA_H
```
Comment (rockdaboot) [inline tests/test-is-cookie-domain-acceptable.c:134]: I think we can just drop using alloca() and always use malloc.
If you don't mind, I'll prepare a patch for it and put you in as `Reported-by:`.

**Comment by rockdaboot** [comment]
@rsbeckerca Thanks for your contribution !
If we can get in #228, there should be no need to include network header files.
I'll come back to here once #228 has been closed or merged.

File: tests/test-is-cookie-domain-acceptable.c (hunk 1 above)
Comment (rsbeckerca) [reply tests/test-is-cookie-domain-acceptable.c:134]: Works for me.

**Comment by rsbeckerca** [comment]
> @rsbeckerca Thanks for your contribution ! If we can get in #228, there should be no need to include network header files. I'll come back to here once #228 has been closed or merged.

Let me know when you are ready. I can reroll the PR if you want.

**Comment by rockdaboot** [comment]
@rsbeckerca Removal of alloca: #231
Please drop that alloca commit from your PR (feel free to force-push).

**Comment by rsbeckerca** [comment]
> @rsbeckerca Removal of alloca: #231 Please drop that alloca commit from your PR (feel free to force-push).

Should be done. I repushed the original commit dropping e85cddd.

**Comment by rockdaboot** [comment]
@rsbeckerca Can you try to build and run tests from latest master without any changes?

**Comment by rsbeckerca** [comment]
> @rsbeckerca Can you try to build and run tests from latest master without any changes?

No, there are a few patches I have to apply to make this work. The summary of what I have to do from the tarball is as follows:
```
	        	sed "/<ws2tcpip.h>/a #elif defined __TANDEM" -i src/psl.c
	        	sed "/defined __TANDEM/a # include <unistd.h>" -i src/psl.c
	        	sed "/defined __TANDEM/a # include <netinet/in.h>" -i src/psl.c
	        	sed "/defined __TANDEM/a # include <netinet/in6.h>" -i src/psl.c
	        	sed "/defined __TANDEM/a # include <netdb.h>" -i src/psl.c
	        	sed "/defined __TANDEM/a # include <sys/socket.h>" -i src/psl.c

	        	sed "s/alloca(/(char *)malloc(/g" -i tests/test-is-cookie-domain-acceptable.c
	        	sed "s/alloca(/(char *)malloc(/g" -i tests/test-is-public-all.c
	        	sed "s/alloca(/(char *)malloc(/g" -i tests/test-is-public-builtin.c
	        	sed "s/alloca(/(char *)malloc(/g" -i tests/test-is-public.c
	        	sed "s/alloca(/(char *)malloc(/g" -i tests/test-registrable-domain.c
```

**Comment by rockdaboot** [comment]
> > @rsbeckerca Can you try to build and run tests from latest master without any changes?
> 
> No, there are a few patches I have to apply to make this work. The summary of what I have to do from the tarball is as follows:
> 
> ```
> 	        	sed "/<ws2tcpip.h>/a #elif defined __TANDEM" -i src/psl.c
> 	        	sed "/defined __TANDEM/a # include <unistd.h>" -i src/psl.c
> 	        	sed "/defined __TANDEM/a # include <netinet/in.h>" -i src/psl.c
> 	        	sed "/defined __TANDEM/a # include <netinet/in6.h>" -i src/psl.c
> 	        	sed "/defined __TANDEM/a # include <netdb.h>" -i src/psl.c
> 	        	sed "/defined __TANDEM/a # include <sys/socket.h>" -i src/psl.c
> 
> 	        	sed "s/alloca(/(char *)malloc(/g" -i tests/test-is-cookie-domain-acceptable.c
> 	        	sed "s/alloca(/(char *)malloc(/g" -i tests/test-is-public-all.c
> 	        	sed "s/alloca(/(char *)malloc(/g" -i tests/test-is-public-builtin.c
> 	        	sed "s/alloca(/(char *)malloc(/g" -i tests/test-is-public.c
> 	        	sed "s/alloca(/(char *)malloc(/g" -i tests/test-registrable-domain.c
> ```

Are you sure you are using latest git master? Because all usages of `alloca()` have been replaced/removed, as well as all the network related include directives.

**Comment by rsbeckerca** [comment]
I can build libpsl.a from the above tarball. Thanks. configure.ac needs a little tweeking to get a .so file, but I don't have a delivery requirement for that, so it is fine as is for now.

**Comment by rockdaboot** [comment]
> I can build libpsl.a from the above tarball. Thanks. configure.ac needs a little tweeking to get a .so file, but I don't have a delivery requirement for that, so it is fine as is for now.

Thanks for testing. Then I close this issue and #229.

**Pull request closed without merging**